internal/controller/prefectserver_controller.go

```diff
@@ -224,56 +226,45 @@ func (r *PrefectServerReconciler) reconcileDeployment(ctx context.Context, serve
 
 	objName := constants.Deployment
 
-	serverNamespacedName := types.NamespacedName{
-		Namespace: server.Namespace,
-		Name:      server.Name,
+	deploy := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      server.Name,
+			Namespace: server.Namespace,
+		},
 	}
 
-	foundDeployment := &appsv1.Deployment{}
-	err = r.Get(ctx, serverNamespacedName, foundDeployment)
-	if errors.IsNotFound(err) {
-		log.Info("Creating Deployment", "name", desiredDeployment.Name)
-		if err = r.Create(ctx, &desiredDeployment); err != nil {
-			condition = conditions.NotCreated(objName, err)
+	mutateFn := func() error {
+		if !metav1.IsControlledBy(&desiredDeployment, server) {
+			errorMessage := fmt.Sprintf(
+				"%s %s already exists and is not controlled by PrefectServer %s",
+				"Deployment", desiredDeployment.Name, server.Name,
+			)
 
-			return &ctrl.Result{}, err
-		}
+			// condition = conditions.AlreadyExists(objName, errorMessage)
 
-		server.Status.Version = prefectiov1.VersionFromImage(desiredDeployment.Spec.Template.Spec.Containers[0].Image)
+			return errors.NewBadRequest(errorMessage)
+		}
 
-		condition = conditions.Created(objName)
-	} else if err != nil {
-		condition = conditions.UnknownError(objName, err)
+		err := mergo.Merge(deploy, desiredDeployment, mergo.WithOverride)
```

Comment:
I had pretty good success by just removing the `desiredThing` function and doing all of that here in the mutate function.  Turned out pretty clean

Reply:
I also think we can just use `meta.SetStatusCondition()` directly and have a single deferred status update for our resource at the end of the function

Reply:
> I had pretty good success by just removing the desiredThing function and doing all of that here in the mutate function. Turned out pretty clean

Ah yeah I'm up for that

> I also think we can just use meta.SetStatusCondition() directly and have a single deferred status update for our resource at the end of the function

Works for me too, may as well not define it elsewhere 👍🏼

Reply:
These are done 👍🏼